app/models/page.rb

```diff
@@ -57,6 +57,24 @@ def meta_tags
     tag_names << plugin_names
   end
 
+  def dup
+    clone = super
+    clone.primary_image = nil
+    clone.slug = nil
+    clone
+  end
+
+  def number_of_pages_with_matching_title
+    Page.where(title: title).count
+  end
+
+  def slug_candidates
+    [
+      :title,
+      [:title, :number_of_pages_with_matching_title]
+    ]
+  end
```

Comment:
A link sounds interesting, but sometimes a page is cloned for testing purposes, and sometimes it's just cloned for convenience (doesn't have anything to do with the parent). So it seems to be shared titles rather than cloned pages that are the stumbling point as far as I can see. Also, you need to be able to tell when editing the page as well as when searching pages which is which at a glance, which I"m not sure the link accomplishes.